server/bin/utils/run-unittest

```diff
@@ -463,7 +463,13 @@ function _setup_state {
         local state
         local user
         while read -r controller name state user; do
-            pbench-state-manager --create=${user} --name=${name} --state=${state}
+            tarball=$(find ${_testdir} -type f -name ${name}.tar.xz)
```

Comment:
Using `find()` here seems like sort of a big hammer.  If we cannot accurately predict the location of the file, perhaps a straightforward glob expression would do?  Something like:
```
            tarball=${_testdir}/*/*/${name}.tar.xz)
```
or whatever is appropriate.

(If you decide to keep the `find()` command, consider reversing the order of the criteria -- I suspect that type-file will be true more often than a name match.)

Reply:
I didn't want to do any more massive legacy test reconstruction than essential. I wish I hadn't had to do this much. Long ago I'd listed the dataset names, and created the `.db` mechanism to pre-load `Dataset` objects... but that didn't include the path. I don't want to go back and analyze all of those test cases to expand the `.db` file with a path; I really don't. The `find` was an expedient fix that could be done once in the script, and I like that.

Reply:
OK.  (I think the glob expression is likely to be more efficient than invoking `find`, but I don't know what constraints this code is facing.)

Given that we're sticking with `find`, do you want to check the `-name` first and then the file `-type`?  I'm figuring that the name will rarely match, which will save a bunch of `stat()` calls trying to decide whether the type matches.